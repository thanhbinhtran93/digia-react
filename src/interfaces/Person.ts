export interface Person {
  name: string;
  email: string;
  phone: string;
}

export interface PersonWithId extends Person {
  id: string;
}
