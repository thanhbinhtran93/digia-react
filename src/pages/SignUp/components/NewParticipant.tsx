/** @jsx jsx */
import { jsx, css } from '@emotion/core';
import React from 'react';
import { useFormik } from 'formik';

import { useParticipantContext } from 'contexts/participantContext';
import { Person } from 'interfaces/Person';
import { participantSchema } from 'vadilators/participant-validator';
import styled from '@emotion/styled';
import { Table } from '../../../components/data-table';
import { Button } from '../../../components/Button';

const getInitialForm = (): Person => {
  return {
    name: '',
    email: '',
    phone: '',
  };
};

const Form = styled('form')`
  width: 100%;
`;

export const NewParticipant: React.FC = (props) => {
  const { addParticipant } = useParticipantContext();

  const formik = useFormik<Person>({
    initialValues: getInitialForm(),
    onSubmit: (values, helpers) => {
      addParticipant(values);
      helpers.resetForm();
    },
    validationSchema: participantSchema,
  });

  return (
    <Form onSubmit={formik.handleSubmit} {...props}>
      {/**TODO: show error */}
      <Table>
        <Table.Row
          css={css`
            padding: 1rem;
          `}
        >
          <Table.Cell>
            <input
              placeholder="Full name"
              name="name"
              value={formik.values.name}
              onChange={formik.handleChange}
            />
          </Table.Cell>
          <Table.Cell>
            <input
              placeholder="E-mail address"
              type="email"
              name="email"
              value={formik.values.email}
              onChange={formik.handleChange}
            />
          </Table.Cell>
          <Table.Cell>
            <input
              placeholder="Phone number"
              type="tel"
              name="phone"
              value={formik.values.phone}
              onChange={formik.handleChange}
            />
          </Table.Cell>
          <Table.Cell>
            <Button type="submit">Add new</Button>
          </Table.Cell>
        </Table.Row>
      </Table>
    </Form>
  );
};
