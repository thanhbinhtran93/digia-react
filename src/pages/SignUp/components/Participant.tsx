/** @jsx jsx */
import { jsx, css } from '@emotion/core';
import React from 'react';
import { PersonWithId } from 'interfaces/Person';
import styled from '@emotion/styled';
import { useParticipantContext } from 'contexts/participantContext';
import { useFormik } from 'formik';
import { participantSchema } from 'vadilators/participant-validator';
import { Button } from '../../../components/Button';
import { Table } from '../../../components/Table';
import { rowCellCss } from '../commonStyles';
import { Input } from 'components/Input';
import { EditIcon, DeleteIcon } from 'components/Icons';

const IconButton = styled.button`
  border: none;
  background: none;
  outline: none;
  padding: 0 0.5rem;
  height: 1.5rem;
  cursor: pointer;
  & > svg {
    fill: #909090;
    width: 1.5rem;
    height: 1.5rem;
  }
`;

interface ParticipantProps {
  participant: PersonWithId;
}

export const Participant: React.FC<ParticipantProps> = ({ participant }) => {
  const [isEditing, setEditing] = React.useState<boolean>(false);

  const toggleEditing = () => setEditing((prev) => !prev);

  if (isEditing) {
    return (
      <EditParticipant
        participant={participant}
        onToggleEditing={toggleEditing}
      />
    );
  }
  return (
    <DisplayParticipant
      participant={participant}
      onToggleEditing={toggleEditing}
    />
  );
};

interface DisplayParticipantProps extends ParticipantProps {
  onToggleEditing: () => void;
}

const DisplayParticipant: React.FC<DisplayParticipantProps> = ({
  participant,
  onToggleEditing,
}) => {
  const { removeParticipant } = useParticipantContext();
  const handleRemove = () => {
    removeParticipant(participant.id);
  };

  return (
    <Table.Row
      key={participant.id}
      css={css`
        border-bottom: 1px solid #f5f5f5;
        padding: 1.5rem;
        line-height: 1.5rem;
        font-weight: 400;
        ${rowCellCss};
      `}
    >
      <Table.Cell>{participant.name}</Table.Cell>
      <Table.Cell>{participant.email}</Table.Cell>
      <Table.Cell>{participant.phone}</Table.Cell>
      <Table.Cell>
        <IconButton onClick={onToggleEditing}>
          <EditIcon />
        </IconButton>
        <IconButton onClick={handleRemove}>
          <DeleteIcon />
        </IconButton>
      </Table.Cell>
    </Table.Row>
  );
};

const Form = styled('form')`
  width: 100%;
`;
interface EditParticipantProps extends ParticipantProps {
  onToggleEditing: () => void;
}

const EditParticipant: React.FC<EditParticipantProps> = ({
  participant,
  onToggleEditing,
  ...props
}) => {
  const { editParticipant } = useParticipantContext();

  const formik = useFormik<PersonWithId>({
    initialValues: participant,
    onSubmit: (values) => {
      editParticipant(values);
      onToggleEditing();
    },
    validationSchema: participantSchema,
  });

  return (
    <Form onSubmit={formik.handleSubmit} {...props}>
      {/**TODO: show error */}
      <Table.Row
        css={css`
          padding: 1rem;
          border-bottom: 1px solid #f5f5f5;
          ${rowCellCss};
        `}
      >
        <Table.Cell>
          <Input
            placeholder="Full name"
            name="name"
            value={formik.values.name}
            onChange={formik.handleChange}
          />
        </Table.Cell>
        <Table.Cell>
          <Input
            placeholder="E-mail address"
            type="email"
            name="email"
            value={formik.values.email}
            onChange={formik.handleChange}
          />
        </Table.Cell>
        <Table.Cell>
          <Input
            placeholder="Phone number"
            type="tel"
            name="phone"
            value={formik.values.phone}
            onChange={formik.handleChange}
          />
        </Table.Cell>
        <Table.Cell>
          <Button color="inverted" onClick={onToggleEditing}>
            Cancel
          </Button>
          <Button color="blue" type="submit">
            Save
          </Button>
        </Table.Cell>
      </Table.Row>
    </Form>
  );
};
