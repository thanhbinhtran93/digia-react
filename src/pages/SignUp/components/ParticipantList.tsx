/** @jsx jsx */
import { jsx, css } from '@emotion/core';
import React from 'react';

import { useParticipantContext } from 'contexts/participantContext';
import { PersonWithId } from 'interfaces/Person';
import { ArrowIcon } from 'components/ArrowIcon';
import { Table } from 'components/Table';
import { useSortableData } from 'hooks/useSortableData';
import { Participant } from './Participant';
import { rowCellCss } from '../commonStyles';

interface Header {
  key: keyof PersonWithId;
  label: string;
}
const headers: Header[] = [
  {
    key: 'name',
    label: 'Name',
  },
  {
    key: 'email',
    label: 'E-mail address',
  },
  {
    key: 'phone',
    label: 'Phone number',
  },
];

export const ParticipantList: React.FC = () => {
  const { participants } = useParticipantContext();
  const { items, sort, currentSortedKey, isSortedDesc } = useSortableData<
    PersonWithId
  >(participants);

  return (
    <Table>
      <Table.RowHeader
        css={css`
          padding: 1rem 1.5rem;
          border-bottom: 1px solid #f5f5f5;
          ${rowCellCss};
        `}
      >
        {headers.map((header) => (
          <Table.Cell key={header.key}>
            <span
              onClick={() => sort({ sortedKey: header.key })}
              css={css`
                cursor: pointer;
                display: flex;
                align-items: center;
                & > svg {
                  height: 1rem;
                }
              `}
            >
              {header.label}
              {currentSortedKey === header.key ? (
                <ArrowIcon direction={isSortedDesc ? 'desc' : 'asc'} />
              ) : null}
            </span>
          </Table.Cell>
        ))}
        <Table.Cell />
      </Table.RowHeader>

      {items.map((participant) => (
        <Participant key={participant.id} participant={participant} />
      ))}
    </Table>
  );
};
