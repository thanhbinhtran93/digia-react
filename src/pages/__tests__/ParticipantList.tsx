import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ParticipantProvider } from 'contexts/participantContext';
import { NewParticipant } from 'pages/SignUp/components/NewParticipant';
import { ParticipantList } from 'pages/SignUp/components/ParticipantList';
import { PersonWithId } from 'interfaces/Person';

it('Edit a row', async () => {
  const { wrapper, editBtn, deleteBtn } = setup();
  expect(wrapper).toMatchSnapshot();

  await userEvent.click(editBtn);
  const saveBtn = wrapper.getByText(/save/i);
  const nameInput = wrapper.getByDisplayValue(/john doe/i);
  const emailInput = wrapper.getByDisplayValue(/john.doe@example.com/i);
  const phoneInput = wrapper.getByDisplayValue(/0441111111/i);

  expect(nameInput).toBeInTheDocument();
  expect(emailInput).toBeInTheDocument();
  expect(phoneInput).toBeInTheDocument();
  expect(screen.getByText(/cancel/i)).toBeInTheDocument();
  expect(saveBtn).toBeInTheDocument();
  expect(editBtn).not.toBeInTheDocument();
  expect(deleteBtn).not.toBeInTheDocument();

  await userEvent.type(nameInput, 'new name');
  await userEvent.click(saveBtn);

  await waitFor(() => {
    expect(screen.getByText('John Doenew name')).toBeInTheDocument();
    expect(screen.getByText('0441111111')).toBeInTheDocument();
  });
});

function setup(props?: React.ComponentProps<typeof ParticipantProvider>) {
  const tree = (
    <ParticipantProvider initialState={initialState} {...props}>
      <NewParticipant />
      <ParticipantList />
    </ParticipantProvider>
  );
  const wrapper = render(tree);
  const editBtn = wrapper.getByTestId('edit-john-doe-id');
  const deleteBtn = wrapper.getByTestId('delete-john-doe-id');

  return { wrapper, editBtn, deleteBtn };
}

const initialState: PersonWithId[] = [
  {
    id: 'john-doe-id',
    name: 'John Doe',
    email: 'john.doe@example.com',
    phone: '0441111111',
  },
  {
    id: 'e732f945-3eed-42d0-9b77-381c232a98dc',
    name: 'Kaycee Cimah',
    email: 'kaycee.cimah@example.com',
    phone: '8468361594',
  },
  {
    id: '2e987437-a4b4-46f5-b93b-ec2e2b5b15a5',
    name: 'Jemimah McLain',
    email: 'jemimah.mclain@example.com',
    phone: '8738048768',
  },
  {
    id: '6232da7f-b973-4ec1-bc1d-efc58b767d1c',
    name: 'Frank Queridas',
    email: 'frank.queridas@example.com',
    phone: '6086265451',
  },
  {
    id: '2a525e5d-3018-42c9-9896-9babe60dc144',
    name: 'Filide Loring',
    email: 'filide.loring@example.com',
    phone: '4784651204',
  },
];
