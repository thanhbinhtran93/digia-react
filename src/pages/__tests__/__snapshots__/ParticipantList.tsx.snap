// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit a row 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="css-1fk24ga-NewParticipant"
      >
        <div
          class="css-qjyi4g-Table"
        >
          <div
            class="css-1m8z7nf-NewParticipant"
          >
            <div
              class="css-195vmrj-Cell"
            >
              <input
                class="css-17f5ri4-Input"
                name="name"
                placeholder="Full name"
                value=""
              />
            </div>
            <div
              class="css-195vmrj-Cell"
            >
              <input
                class="css-17f5ri4-Input"
                name="email"
                placeholder="E-mail address"
                type="email"
                value=""
              />
            </div>
            <div
              class="css-195vmrj-Cell"
            >
              <input
                class="css-17f5ri4-Input"
                name="phone"
                placeholder="Phone number"
                type="tel"
                value=""
              />
            </div>
            <div
              class="css-195vmrj-Cell"
            >
              <button
                class="css-3ox8zq"
                color="primary"
                type="submit"
              >
                Add new
              </button>
            </div>
          </div>
        </div>
      </form>
      <div
        class="css-qjyi4g-Table"
      >
        <div
          class="css-1ub16qz-ParticipantList"
        >
          <div
            class="css-195vmrj-Cell"
          >
            <span
              class="css-m8ub5d"
            >
              Name
            </span>
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <span
              class="css-m8ub5d"
            >
              E-mail address
            </span>
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <span
              class="css-m8ub5d"
            >
              Phone number
            </span>
          </div>
          <div
            class="css-195vmrj-Cell"
          />
        </div>
        <div
          class="css-otiz8p-DisplayParticipant"
        >
          <div
            class="css-195vmrj-Cell"
          >
            John Doe
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            john.doe@example.com
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            0441111111
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <button
              class="css-s6bjet"
              data-testid="edit-john-doe-id"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Edit icon
                </title>
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
            <button
              class="css-s6bjet"
              data-testid="delete-john-doe-id"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Delete icon
                </title>
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="css-otiz8p-DisplayParticipant"
        >
          <div
            class="css-195vmrj-Cell"
          >
            Kaycee Cimah
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            kaycee.cimah@example.com
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            8468361594
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <button
              class="css-s6bjet"
              data-testid="edit-e732f945-3eed-42d0-9b77-381c232a98dc"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Edit icon
                </title>
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
            <button
              class="css-s6bjet"
              data-testid="delete-e732f945-3eed-42d0-9b77-381c232a98dc"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Delete icon
                </title>
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="css-otiz8p-DisplayParticipant"
        >
          <div
            class="css-195vmrj-Cell"
          >
            Jemimah McLain
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            jemimah.mclain@example.com
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            8738048768
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <button
              class="css-s6bjet"
              data-testid="edit-2e987437-a4b4-46f5-b93b-ec2e2b5b15a5"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Edit icon
                </title>
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
            <button
              class="css-s6bjet"
              data-testid="delete-2e987437-a4b4-46f5-b93b-ec2e2b5b15a5"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Delete icon
                </title>
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="css-otiz8p-DisplayParticipant"
        >
          <div
            class="css-195vmrj-Cell"
          >
            Frank Queridas
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            frank.queridas@example.com
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            6086265451
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <button
              class="css-s6bjet"
              data-testid="edit-6232da7f-b973-4ec1-bc1d-efc58b767d1c"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Edit icon
                </title>
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
            <button
              class="css-s6bjet"
              data-testid="delete-6232da7f-b973-4ec1-bc1d-efc58b767d1c"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Delete icon
                </title>
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="css-otiz8p-DisplayParticipant"
        >
          <div
            class="css-195vmrj-Cell"
          >
            Filide Loring
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            filide.loring@example.com
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            4784651204
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <button
              class="css-s6bjet"
              data-testid="edit-2a525e5d-3018-42c9-9896-9babe60dc144"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Edit icon
                </title>
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
            <button
              class="css-s6bjet"
              data-testid="delete-2a525e5d-3018-42c9-9896-9babe60dc144"
            >
              <svg
                fill="currentColor"
                focusable="false"
                height="48"
                role="img"
                viewBox="0 0 24 24"
                width="48"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Delete icon
                </title>
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <form
      class="css-1fk24ga-NewParticipant"
    >
      <div
        class="css-qjyi4g-Table"
      >
        <div
          class="css-1m8z7nf-NewParticipant"
        >
          <div
            class="css-195vmrj-Cell"
          >
            <input
              class="css-17f5ri4-Input"
              name="name"
              placeholder="Full name"
              value=""
            />
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <input
              class="css-17f5ri4-Input"
              name="email"
              placeholder="E-mail address"
              type="email"
              value=""
            />
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <input
              class="css-17f5ri4-Input"
              name="phone"
              placeholder="Phone number"
              type="tel"
              value=""
            />
          </div>
          <div
            class="css-195vmrj-Cell"
          >
            <button
              class="css-3ox8zq"
              color="primary"
              type="submit"
            >
              Add new
            </button>
          </div>
        </div>
      </div>
    </form>
    <div
      class="css-qjyi4g-Table"
    >
      <div
        class="css-1ub16qz-ParticipantList"
      >
        <div
          class="css-195vmrj-Cell"
        >
          <span
            class="css-m8ub5d"
          >
            Name
          </span>
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          <span
            class="css-m8ub5d"
          >
            E-mail address
          </span>
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          <span
            class="css-m8ub5d"
          >
            Phone number
          </span>
        </div>
        <div
          class="css-195vmrj-Cell"
        />
      </div>
      <div
        class="css-otiz8p-DisplayParticipant"
      >
        <div
          class="css-195vmrj-Cell"
        >
          John Doe
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          john.doe@example.com
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          0441111111
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          <button
            class="css-s6bjet"
            data-testid="edit-john-doe-id"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Edit icon
              </title>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
          <button
            class="css-s6bjet"
            data-testid="delete-john-doe-id"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Delete icon
              </title>
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="css-otiz8p-DisplayParticipant"
      >
        <div
          class="css-195vmrj-Cell"
        >
          Kaycee Cimah
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          kaycee.cimah@example.com
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          8468361594
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          <button
            class="css-s6bjet"
            data-testid="edit-e732f945-3eed-42d0-9b77-381c232a98dc"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Edit icon
              </title>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
          <button
            class="css-s6bjet"
            data-testid="delete-e732f945-3eed-42d0-9b77-381c232a98dc"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Delete icon
              </title>
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="css-otiz8p-DisplayParticipant"
      >
        <div
          class="css-195vmrj-Cell"
        >
          Jemimah McLain
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          jemimah.mclain@example.com
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          8738048768
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          <button
            class="css-s6bjet"
            data-testid="edit-2e987437-a4b4-46f5-b93b-ec2e2b5b15a5"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Edit icon
              </title>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
          <button
            class="css-s6bjet"
            data-testid="delete-2e987437-a4b4-46f5-b93b-ec2e2b5b15a5"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Delete icon
              </title>
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="css-otiz8p-DisplayParticipant"
      >
        <div
          class="css-195vmrj-Cell"
        >
          Frank Queridas
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          frank.queridas@example.com
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          6086265451
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          <button
            class="css-s6bjet"
            data-testid="edit-6232da7f-b973-4ec1-bc1d-efc58b767d1c"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Edit icon
              </title>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
          <button
            class="css-s6bjet"
            data-testid="delete-6232da7f-b973-4ec1-bc1d-efc58b767d1c"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Delete icon
              </title>
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="css-otiz8p-DisplayParticipant"
      >
        <div
          class="css-195vmrj-Cell"
        >
          Filide Loring
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          filide.loring@example.com
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          4784651204
        </div>
        <div
          class="css-195vmrj-Cell"
        >
          <button
            class="css-s6bjet"
            data-testid="edit-2a525e5d-3018-42c9-9896-9babe60dc144"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Edit icon
              </title>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04a.996.996 0 000-1.41l-2.34-2.34a.996.996 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
          <button
            class="css-s6bjet"
            data-testid="delete-2a525e5d-3018-42c9-9896-9babe60dc144"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="48"
              role="img"
              viewBox="0 0 24 24"
              width="48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Delete icon
              </title>
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
